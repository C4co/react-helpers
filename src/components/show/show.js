import React, { Fragment } from "react"
import PropTypes from "prop-types"

export function Show(props) {
  if (props.if) {
    return (
      <Fragment>
        {props.children}
      </Fragment>
    )
  }

  return null
}

Show.propTypes = {
  /** Condition to show or hide the component. */
  if: PropTypes.bool,

  /** Children content */
  children: PropTypes.any
}

Show.defaultProps = {
  if: true
}
