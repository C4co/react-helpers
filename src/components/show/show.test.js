import React from "react"
import { mount } from "enzyme"
import { Show } from "./show"

describe("Show component", () => {
  test("Check visibility", () => {
    const visible1 = mount(
      <Show if>
        <h1>i am visible</h1>
      </Show>
    )

    // undefined don't set default {if} propertie.
    const visible2 = mount(
      <Show if={undefined}>
        <h1>i am visible</h1>
      </Show>
    )

    expect(visible1.contains(<h1>i am visible</h1>)).toBe(true)
    expect(visible1).toMatchSnapshot()

    expect(visible2.contains(<h1>i am visible</h1>)).toBe(true)
    expect(visible2).toMatchSnapshot()
  })

  test("Check hidden", () => {
    const hidden1 = mount(
      <Show if={false}>
        <h1>i am hidden</h1>
      </Show>
    )

    const hidden2 = mount(
      <Show if={null}>
        <h1>i am hidden</h1>
      </Show>
    )

    expect(hidden1.contains(<h1>i am hidden</h1>)).toBe(false)
    expect(hidden1).toMatchSnapshot()

    expect(hidden2.contains(<h1>i am hidden</h1>)).toBe(false)
    expect(hidden2).toMatchSnapshot()
  })
})
