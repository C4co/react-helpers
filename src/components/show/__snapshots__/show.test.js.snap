// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show component Check hidden 1`] = `
<Show
  if={false}
/>
`;

exports[`Show component Check hidden 2`] = `
<Show
  if={null}
/>
`;

exports[`Show component Check visibility 1`] = `
<Show
  if={true}
>
  <h1>
    i am visible
  </h1>
</Show>
`;

exports[`Show component Check visibility 2`] = `
<Show
  if={true}
>
  <h1>
    i am visible
  </h1>
</Show>
`;
